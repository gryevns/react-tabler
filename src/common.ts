export type Color =
    | "blue"
    | "azure"
    | "indigo"
    | "purple"
    | "pink"
    | "red"
    | "orange"
    | "yellow"
    | "lime"
    | "green"
    | "teal"
    | "cyan"
    | "gray";

export const colors = [
    "blue" as const,
    "azure" as const,
    "indigo" as const,
    "purple" as const,
    "pink" as const,
    "red" as const,
    "orange" as const,
    "yellow" as const,
    "lime" as const,
    "green" as const,
    "teal" as const,
    "cyan" as const,
    "gray" as const,
];

export type Variant =
    | "primary"
    | "secondary"
    | "success"
    | "info"
    | "warning"
    | "danger";

export const variants = [
    "primary" as const,
    "secondary" as const,
    "success" as const,
    "info" as const,
    "warning" as const,
    "danger" as const,
];

export type Size = "sm" | "md" | "lg" | "xl";

export const sizes = ["sm" as const, "md" as const, "lg" as const, "xl" as const];
