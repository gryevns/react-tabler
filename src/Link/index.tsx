import React from "react";
import cn from "classnames";

import { Variant } from "../common";

interface LinkProps {
    children: React.ReactNode;
    variant?: Variant;
}

const Link = React.forwardRef(
    (
        {
            variant = "primary",
            ...props
        }: LinkProps & React.AnchorHTMLAttributes<HTMLElement>,
        ref: React.Ref<HTMLAnchorElement>,
    ) => {
        const classes = cn(`link-${variant}`);
        return (
            <a ref={ref} {...props} className={classes}>
                {props.children}
            </a>
        );
    },
);

export default Link;
