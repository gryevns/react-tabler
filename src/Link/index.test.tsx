import React from "react";
import { shallow } from "enzyme";

import Link from "./index";
import { variants } from "../common";

describe("Link", () => {
    it("should match snapshot", () => {
        variants.map(variant => {
            const wrapper = shallow(<Link variant={variant}>link</Link>);
            expect(wrapper).toMatchSnapshot();
        });
    });
});
