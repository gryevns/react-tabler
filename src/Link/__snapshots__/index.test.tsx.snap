// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should match snapshot 1`] = `
<a
  className="link-primary"
>
  link
</a>
`;

exports[`Link should match snapshot 2`] = `
<a
  className="link-secondary"
>
  link
</a>
`;

exports[`Link should match snapshot 3`] = `
<a
  className="link-success"
>
  link
</a>
`;

exports[`Link should match snapshot 4`] = `
<a
  className="link-info"
>
  link
</a>
`;

exports[`Link should match snapshot 5`] = `
<a
  className="link-warning"
>
  link
</a>
`;

exports[`Link should match snapshot 6`] = `
<a
  className="link-danger"
>
  link
</a>
`;
